src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/organisations/OrganisationsEventListener.kt

```diff
@@ -44,6 +44,7 @@ class OrganisationsEventListener(
               "ADDRESSES_CORPORATE-INSERTED" -> synchronisationService.corporateAddressInserted(sqsMessage.Message.fromJson())
               "ADDRESSES_CORPORATE-UPDATED" -> synchronisationService.corporateAddressUpdated(sqsMessage.Message.fromJson())
               "ADDRESSES_CORPORATE-DELETED" -> synchronisationService.corporateAddressDeleted(sqsMessage.Message.fromJson())
+              "PHONES_CORPORATE-INSERTED" if sqsMessage.Message.asCorporatePhoneEvent().isAddress -> synchronisationService.corporateAddressPhoneInserted(sqsMessage.Message.fromJson())
```

Comment:
guard is experimental; so could move the `if` to right of `when` clause if it becomes an issue.

Reply:
Interesting .. though if it needs a compiler option, is this standard kotlin?